h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -720,14 +722,26 @@ public final double likelihood(double yr, double ym) {
 
     public GLMWeights computeWeights(double y, double eta, double off, double w, GLMWeights x) {
       double etaOff = eta + off;
-      x.mu = linkInv(etaOff);
+      x.mu = linkInv(etaOff); // prob estimated response = 1
       double var = variance(x.mu);//Math.max(1e-5, variance(x.mu)); // avoid numerical problems with 0 variance
       double d = linkDeriv(x.mu);
       x.w = w / (var * d * d);
       x.z = eta + (y - x.mu) * d;
       likelihoodAndDeviance(y,x,w);
       return x;
     }
+
+    public GLMWeights computeWeightsCOD(double y, double eta, double off, double w, GLMWeights x) {
+      double etaOff = eta + off;
+      x.mu = linkInv(etaOff); // prob estimated response = 1
+      double temp = x.mu*(1-x.mu);
+      x.respMinusPr = w*(y-x.mu);
+      x.prOneMinpr = w*(temp < 1e-6?1e-6:temp);
+      x.w = x.prOneMinpr;
+      x.z = x.respMinusPr;
+      likelihoodAndDeviance(y,x,w);
```

Comment:
I am adding this function instead of using the old one because it is simplier and more numerically stable. For example, x.w is equivalent to x.mu*(1-x.mu). However, it is calculated as w/(var * d * d) which can be simplified to x.mu * (1-x.mu) for Binomials only.